lstchain/scripts/lstchain_find_pedestals.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+
+"""
+This script reads all DL1 subrun files (i.e. not merged) in the input
+directory, and finds the event_id's of interleaved pedestals. It writes them
+out into one hdf5 file for each sub-run, called pedestal_ids_RunXXXXX.YYYY.h5,
+which contains a single table with one column, 'event_id'
+"""
+
+import argparse
+import numpy as np
+import pandas as pd
+
+from pathlib import Path
+from ctapipe.io import read_table
+from lstchain.paths import parse_dl1_filename
+
+parser = argparse.ArgumentParser(description="Interleaved Pedestal Finder")
+
+parser.add_argument('-d', '--input-dir', dest='srcdir',
+                    type=Path, default='./',
+                    help='path to the directory of the DL1 files'
+                    )
+
+parser.add_argument('output_dir', metavar='output-dir', type=Path,
+                    help='Path where to store the output hdf5 files'
+                    )
+
+def main():
+
+    args = parser.parse_args()
+
+    files = sorted(args.srcdir.glob('dl1_LST-1.Run?????.????.h5'))
+    if not files:
+        raise IOError("No input dl1 files found")
+
+    output_dir = args.output_dir.absolute()
+    output_dir.mkdir(exist_ok=True, parents=True)
+
+    for file in files:
+        run_info = parse_dl1_filename(file)
+        run_number, subrun_index = run_info.run, run_info.subrun
+
+        # Approximate interleaved pedestal frequency in Hz:
+        approximate_frequency = 50
+        if run_number > 2708:
+            approximate_frequency = 100
+
+        table = read_table(file, '/dl1/event/telescope/parameters/LST_LSTCam')
+        all_times = np.array(table['dragon_time'])
+        # probable interleaved flat-field events (just in case tag is
+        # faulty!):
+        ffmask = ((table['intensity'] > 3e4) &
+                  (table['concentration_pixel'] < 0.005))
+
+        pedmask = np.zeros(len(table), dtype=bool)
+        pedmask[~ffmask] = find_pedestals(all_times[~ffmask],
+                                          approximate_frequency)
+
+        # Now remove the 10 brightest events (might be cosmics accidentally
+        # falling in the time windows determined by time_pedestals). the
+        # expected value of cosmics is smaller that, so we are probably
```

Comment:
Incomplete sentence?

Reply:
?? It is finished below: "we are probably removing some pedestals, but it does not harm."

Reply:
I meant "The expected value of cosmics is smaller that, so ..." (smaller than?)